hail-ci-build.sh

```diff
@@ -1,7 +1,38 @@
-set -x
-source activate hail
-GRADLE_OPTS=-Xmx2048m ./gradlew testAll makeDocs archiveZip --gradle-user-home /gradle-cache
+#!/bin/bash
+set -ex
+
+CLUSTER_NAME=ci-test-${SOURCE_SHA:0:12}-${SOURCE_SHA:0:12}
+
+time source activate hail
+time pip search cloudtools
+time pip install -U cloudtools
+gcloud auth activate-service-account \
+    hail-ci-0-1@broad-ctsa.iam.gserviceaccount.com \
+    --key-file=/secrets/hail-ci-0-1.key && \
+
+shutdown_cluster() {
+    set +e
+    time cluster stop --async ${CLUSTER_NAME}
+    exit
+}
+trap shutdown_cluster INT TERM
+
+GRADLE_OPTS=-Xmx2048m ./gradlew testAll makeDocs archiveZip --gradle-user-home /gradle-cache && \
```

Comment:
With `set -ex`, these && and trailing backslashes aren't necessary (and a bit more readable).  Also, the EXIT_CODE can go, too.